Automattic/wp-calypso
# useSiteMetricsQuery: add support to disable automated data fetching

Related to #93032

## Proposed Changes

`useSiteMetricsQuery` runs Tanstack `useQuery` under the hood. This hook offers an `enabled` prop. This PR adds the option to pass values to that prop when calling `useSiteMetricsQuery`

I'd recommend reviewing this PR one commit at a time. The first commit on this branch adds the new prop, and implements it in the new bandwidth section of the hosting overview. The second commit updates the function to accept a single object instead of individual arguments. I didn't like the new unnamed boolean, and thought this would improve the ergonomics of the function. Other usages are also updated in the same commit. If we'd rather not make this change to the format of the arguments, we can revert just that one commit and go back to individual arguments.

## Why are these changes being made?

`useQuery`'s `enabled` prop defaults to `true`, which allows the query to automatically refetch under a number of different circumstances, including re-renders. If set to `false` the query doesn't run automatically, which is helpful if you want to delay or stop automatic fetching/refetching. We'll leave it defaulting to true, so most implementations won't need to worry about it, but for cases where disabling the query is helpful, we'll have a means of accomplishing it.

## Testing Instructions

- Visit the Calypso Live link for this branch
- Open /overview for any of your sites by clicking on the Hosting Overview button
- Append ?flags=hosting-overview-refinements to the end of the url
- Confirm that the new Bandwidth data loads as expected, and that it doesn't refresh or re-render its placeholder after the data appears for the first time
- Visit the Monitoring tab of the hosting overview and confirm that all of the charts load as expected with accurate data.
- Check for any new errors in the console
